crates/context_aware_config/src/helpers.rs

```diff
@@ -167,15 +177,15 @@ pub fn validate_jsonschema(
 pub fn calculate_context_priority(
     _object_key: &str,
     cond: &Value,
-    dimension_schema_map: &HashMap<String, (JSONSchema, i32)>,
+    dimension_schema_map: &HashMap<String, DimensionData>,
 ) -> Result<i32, String> {
     let get_priority = |key: &str, val: &Value| -> Result<i32, String> {
         if key == "var" {
             let dimension_name =
                 val.as_str().ok_or("failed to decode dimension as str")?;
             dimension_schema_map
                 .get(dimension_name)
-                .map(|(_, priority)| priority)
+                .map(|dimension_val| &dimension_val.priority)
```

Comment:
```suggestion
                .map(|dimension_val| &dimension_val.weight)
```

Reply:
@Datron this function calculates priority which is currently supported , 
for weight calculation there is different function `calculate_context_weightage`